### ISSUE TITLE
Prune/Freeze/FineTune

### ISSUE THREAD

**Issue by SvenMantowsky** [lmbxmu/HRankPlus#3]
First of all - very nice work!
I just wonder, if i missed something but i thought a bunch of filters get pruned, some weights get frozen and the other ones get finetune (updated) in the training afterwards.
But if i check the weights after pruning the filters with
for param in model.parameters():
    print(param.requires_grad)
then every parameter returns True --> is updated in the training afterwards. 

could you point out my mistake here, what i overlooked.
Thanks in advance

**Comment by lmbxmu** [comment]
Thanks for your interest.

The code provided here fine-tunes all preserved filters. If you want to freeze specific filters, you need to modify the code by unabling the gradient of these filters to pass back.

**Comment by SvenMantowsky** [comment]
Just one last question: is there a specific reason why you used svd as a ranking metric instead of, e.g. the mean of the filter or something else?

**Comment by lmbxmu** [comment]
Sorry. It is a big misunderstanding.

We did not use svd as a ranking metric. Instead, we use the rank (linear algebra) of the feature map to measure the relative importance of each filter.  The svd in the paper was trying to explain that high-rank feature map contains more information thus the corresponding filter should be preserved.

**Comment by SvenMantowsky** [comment]
In the get_feature_hook() function you use torch.rank_matrix() to get the imprtance of the filter an after that you sort them by their value if i understood the code correct. Maybe my question was misleading. So maybe i can make it clearer. 
Why did you use torch.rank_matrix() and not for example mean() or somethkng? 
Sorry for all the inconvenience.

**Comment by lmbxmu** [comment]
Why don't you have a look at the paper.

It is a boring question and I choose not to answer.

**Comment by SvenMantowsky** [comment]
I read your paper and there is no answer to this. You only mentioned svd twice in a really short paragraph, thats the reason why i asked. 
the reason that it is less dependent on the image distribution P (I) was just too imprecise for me.

**Comment by lmbxmu** [comment]
I think you haven't even understood what this paper has done so far.

**Comment by SvenMantowsky** [comment]
If you wont answer my question nevermind. Pruning algroithms are not really complex, so its not that hard to understand them. i just wanted to know why you used svd instead of another linear algebra method to calculate the information content. Still nice work anf thanks for answering

**Comment by www516717402** [comment]
@SvenMantowsky This is base knowledge in linear Algebra.
